src/ex2/main.js

```diff
@@ -1,10 +1,86 @@
+import ItemManager from "./item_manager.js";
 // Implement the `Main` class here
+class Main {
+	constructor() {
+		this.itemManager = new ItemManager();
+	}
+	init() {
+		document
+			.getElementById("list-item-submit")
+			.addEventListener("click", async () => {
+				await this.validateInput();
+			});
+		document.getElementById("clear-btn").addEventListener("click", () => {
+			this.clearAll();
+		});
+	}
 
+	renderTaskList(itemArray) {
+		const tasksList = document.getElementById("list");
+		const empyState = document.getElementById("empty-state");
+		const clearBtn = document.getElementById("clear-btn");
+		const inputTxt = document.getElementById("list-item-input");
+		tasksList.innerHTML = "";
+		if (itemArray.length > 0) {
+			itemArray.forEach((item) => {
+				empyState.style.display = "none";
+				const newItem = document.createElement("li");
+				newItem.innerText = item.value;
+				newItem.id = item.id;
+				newItem.className = "list-item";
+
+				const trashImg = document.createElement("img");
+				trashImg.src = "./images/delete_icon.svg";
+				trashImg.className = "list-item-delete-button";
+				trashImg.addEventListener("click", (e) => {
+					e.stopPropagation();
+					this.itemManager.removeItem(newItem.id);
+					newItem.remove();
+					this.displayEmptyState();
+				});
+				newItem.appendChild(trashImg);
+				tasksList.appendChild(newItem);
+				clearBtn.style.display = "block";
+				inputTxt.value = "";
+			});
+		} else {
+			this.displayEmptyState();
+		}
+	}
+
+	displayEmptyState() {
+		const tasksList = document.getElementById("list");
+		const empyState = document.getElementById("empty-state");
+		const clearBtn = document.getElementById("clear-btn");
+
+		if (tasksList.childElementCount === 0) {
+			clearBtn.style.display = "none";
+			empyState.style.display = "block";
+		}
+	}
+
+	async validateInput() {
```

Comment:
This function is handling the input but not validating it.. validating would be to check if the user added the same item twice for example (one of the bonuses).